### ISSUE TITLE
nansum with complex support

### ISSUE THREAD

**Issue by PatrikBrynolfsson** [pytorch/pytorch#115164]
### 🚀 The feature, motivation and pitch

I am missing support for complex input in nansum(). nansum() is the basis for many of the nan* functions such as nanmean(), nanstd() etc, so missing support for complex is problematic i think. 

### Alternatives

This is a workaround:
torch.sum(z[z.isnan().logical_not()])
but having nansum() would be a lot cleaner.

**Comment by janeyx99** [comment]
also cc @cpuhrsch if you have thoughts on the nan* APIs.

**Comment by cpuhrsch** [comment]
@janeyx99 Well, of course I'd advocate for further extension of https://pytorch.org/docs/master/masked.html . This also solves the autograd issue and can be used for more than NaNs. It might even already work for this case.
